elasticdl/python/worker/task_data_service.py

```diff
@@ -153,11 +154,22 @@ def get_dataset(self):
             # sure `read_records()` is executed without iterating all the
             # records so this should not be time consuming.
             if self._warm_up_task is None and not self._has_warmed_up:
-                task = self._worker.get_task()
-                self._warm_up_task = task
-                for _ in self.data_reader.read_records(task):
-                    break
-                self._has_warmed_up = True
+                while True:
+                    task = self._worker.get_task()
+                    if task.type != elasticdl_pb2.WAIT:
```

Comment:
Will the program hang here forever?

Reply:
No, this is the same logic as before (without relaunch).
(1) the master will stop sending `WAIT` task after doing tasks are done. 
(2) In the unlikely event that the master fails, `get_task` RPC call will fail the worker pod.

Reply:
Perhaps it would be good to add some comments on this logic?